app/src/org/commcare/tasks/ProcessAndSendTask.java

```diff
@@ -194,6 +195,13 @@ private boolean checkFormRecordStatus(FormRecord[] records)
                         //Otherwise, the SD card just got removed, and we need to bail anyway.
                         throw e;
                     }
+                } catch(InvalidStorageStructureException e) {
+                    // Thrown when updating a case that isn't present
+                    // Do same behavior as InvalidStructureException (wipe record)
+                    CommCareApplication._().reportNotificationMessage(NotificationMessageFactory.message(ProcessIssues.BadTransactions), true);
+                    Logger.log(AndroidLogger.TYPE_ERROR_DESIGN, "Removing form record due to case transaction data|" + getExceptionText(e));
+                    FormRecordCleanupTask.wipeRecord(c, record);

```

Comment:
What if we quarantined these forms (`processor.updateRecordStatus(record, FormRecord.STATUS_LIMBO);`)? If a user reports a form not making it to HQ, it might be helpful to be able to see which ones on the device so they can copy over data to a new form. Also possible that this is a sync related bug and a 412 sync would allow them to recover the form to a state that they can successfully submit it.